src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/ModuleBuilderImpl.cs

```diff
@@ -172,9 +164,15 @@ private AssemblyReferenceHandle GetAssemblyReference(Assembly assembly)
 
         internal EntityHandle GetTypeHandle(Type type)
         {
-            if (type is TypeBuilderImpl tb && _typeDefinitions.TryGetValue(tb, out var handle))
+            if (type is TypeBuilderImpl tb && Equals(tb.Module))
             {
-                return handle;
+                foreach(TypeDefinitionWrapper typeDef in _typeDefinitions)
```

Comment:
`TypeBuilderImpl` should have a field with its `TypeDefinitionHandle`. `TypeDefinitionWrapper` and this look should not be necessary.

While you are on it, you can also implement `TypeBuilderImpl.MetadataToken` to return the value of `TypeDefinitionHandle`.

Reply:
Sure, thank you!